### PR TITLE
Redis stuff

### PR DESCRIPTION
**Pull request by jooleyaz**
Added functionality for invite codes, creating should work with testing and joining has not been tested yet (need to wait until we have a "create invite code" page)

### CODE REVIEW

File: .expo/README.md
Comment (cameronl10) [inline .expo/README.md:1]: what is this?

File: .expo/settings.json
Comment (cameronl10) [inline .expo/settings.json:1]: what is this?

File: backend/src/handlers/householdHandler.ts [hunk 1]
```diff
@@ -1,4 +1,16 @@
+import { stringify } from "querystring";
```
Comment (cameronl10) [inline backend/src/handlers/householdHandler.ts]: what is this?

File: frontend/managers/joinCodes.ts
Comment (cameronl10) [inline frontend/managers/joinCodes.ts:1]: what is this?

File: package-lock.json
Comment (cameronl10) [inline package-lock.json:1]: what is this?

File: backend/src/handlers/householdHandler.ts [hunk 2]
```diff
@@ -127,4 +138,35 @@ async function getBills(houseid: string): Promise<Bill[]> {
     }
 };
 
-export { getHousehold, getHouseholdByUser, createHousehold, editHousehold, deleteHousehold, getWishGroups, getUsers, getBills };
+// create an 8-digit alphanumerical random join code as key,
```
Comment (jaykbpark) [inline backend/src/handlers/householdHandler.ts:141]: why is this 8? should be 6 LMAO

File: frontend/api/client.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { GraphQLClient } from 'graphql-request'
 import { getSessionKey } from '@/managers/sessionManager';
-const API_URL = 'http://localhost:3000/graphql';
+const API_URL = 'http://18.220.131.95:3000/graphql';
```
Comment (jaykbpark) [inline frontend/api/client.ts]: remove please 💀

File: frontend/app/createHouse.tsx [hunk 4]
```diff
@@ -30,17 +34,32 @@ const CreateHouse = () => {
...
+        const checkHouseCodeQuery = useQuery({
+          queryKey: [cleanFormInput],
+          queryFn: async ({queryKey}) => {
+            await checkHouseCode(queryKey);
+          }
+        });
+        const {data} = useSessionInfo();
+        if (checkHouseCodeQuery.isSuccess) {
+          await addUser(data.userid, checkHouseCodeQuery.data);
+          alert("Success!! :3");
```
Comment (jaykbpark) [inline frontend/app/createHouse.tsx]: remove :3

File: frontend/app/createHouse.tsx [hunk 5]
```diff
@@ -30,17 +34,32 @@ const CreateHouse = () => {
...
-      const otpString = otpValue.join('');
-      alert(otpString);
-    } else {
-      setOtpFilled(false);
+  
+  const onJoinSubmit = async (joinData: JoinGroupFormData) => {
+      if (otpValue.every((value) => value !== '')) { 
+        setOtpFilled(true);
+        const otpString = otpValue.join('');
+        alert(otpString);
```
Comment (jaykbpark) [inline frontend/app/createHouse.tsx:46]: are all these alerts necessary

Comment (jooleyaz) [reply frontend/app/createHouse.tsx:46]: this was erik's approved changes idk

**Review by cameronl10 — APPROVED** [review]
lgtm

**Pull request merged**